helper/sign_attachment.go

```diff
@@ -30,21 +30,24 @@ func EncryptSignAttachment(
 		return nil, nil, nil, err
 	}
 
-	if unlockedKeyObj, err = privateKeyObj.Unlock(passphrase); err != nil {
+	if packets, err = publicKeyRing.EncryptAttachment(binMessage, fileName); err != nil {
 		return nil, nil, nil, err
 	}
 
-	if privateKeyRing, err = crypto.NewKeyRing(unlockedKeyObj); err != nil {
+	if unlockedKeyObj, err = privateKeyObj.Unlock(passphrase); err != nil {
 		return nil, nil, nil, err
 	}
 
-	if packets, err = publicKeyRing.EncryptAttachment(binMessage, fileName); err != nil {
+	if privateKeyRing, err = crypto.NewKeyRing(unlockedKeyObj); err != nil {
+		unlockedKeyObj.ClearPrivateParams()
 		return nil, nil, nil, err
 	}
 
 	if signatureObj, err = privateKeyRing.SignDetached(binMessage); err != nil {
+		unlockedKeyObj.ClearPrivateParams()
 		return nil, nil, nil, err
 	}
+	unlockedKeyObj.ClearPrivateParams()
```

Comment:
This is pretty nitpicky but since it's kinda cool I thought I'd mention it anyway:

Instead of repeating the `ClearPrivateParams` three times, you can do:
```go
	if unlockedKeyObj, err = privateKeyObj.Unlock(passphrase); err != nil {
		return nil, nil, nil, err
	}
	defer unlockedKeyObj.ClearPrivateParams()
```
to automatically call it when the function returns.

Maybe this is the way we should recommend to use this function in the readme and stuff :)

Reply:
Sliding in to raise a question: Shouldn't `KeyRing.ClearPrivateParams` return an `ok bool`, as `Key.ClearPrivateParams`? 
In which case, usage could be recommended as
```
defer func() {
    if !unlockedKeyObj.ClearPrivateParams() {
        panic("Couldn't clear private params!")
    }
}()
```
(or something more appropriate)

Reply:
If it returns false it means there was nothing more to clear, and IMHO is nothing worth panicking

I will introduce the defer and merge :)